docs/topics/yaml-rule-structure-syntax.adoc

```diff
@@ -0,0 +1,520 @@
+
+Currently, the analyzer adds the following labels to dependencies:
+
+[source,terminal]
+----
+labels:
+- konveyor.io/dep-source=internal
+- konveyor.io/language=java
+----
+
+.Dependency label selector
+
+The analyzer CLI accepts the `--dep-label-selector` option, which allows filtering-in or filtering-out incidents generated from a dependency by their labels.
+
+For example, the analyzer adds a `konveyor.io/dep-source` label to dependencies with a value that indicates whether the dependency is a known open-source dependency.
+
+To exclude incidents for all such open-source dependencies, you can use `--dep-label-selector` as follows:
+
+`konveyor-analyzer ... --dep-label-selector !konveyor.io/dep-source=open-source`
+
+The Java provider in the analyzer can also add an exclude label to a list of packages. To exclude all such packages, you can use `--dep-label-selector` as follows:
+
+`konveyor-analyzer ... --dep-label-selector !konveyor.io/exclude`
+
+[id="yaml-rule-categories_{context}"]
+=== Rule categories
+
+* `mandatory`
+** You must resolve the issue for a successful migration, otherwise, the resulting application is not expected to build or run successfully. An example of such an issue is proprietary APIs that are not supported on the target platform.
+* `optional`
+** If you do not resolve the issue, the application is expected to work, but the results might not be optimal. If you do not make the change at the time of migration, you need to put it on the schedule soon after your migration is completed. An example of such an issue is EJB 2.x code not upgraded to EJB 3.
+* `potential`
+** You need to examine the issue during the migration process, but there is not enough information to determine whether resolving the issue is mandatory for the migration to succeed. An example of such an issue is migrating a third-party proprietary type when there is no directly compatible type on the target platform.
+
+
+[id="yaml-rule-actions_{context}"]
+=== Rule Actions
+
+Rules can include 2 types of actions: _message_ and _tag_. Each rule includes one of them or both.
+
+.Message actions
+
+The message action creates an issue with a message when the rule matches. The custom data exported by providers can also be used in the message.
+
+`message: "helpful message about the issue"`
+
+_Example:_
+
+[source,terminal]
+----
+- ruleID: test-rule
+  when:
+    <CONDITION>
+  message: Test rule matched. Please resolve this migration issue.
+----
+
+Optionally, a message can include hyperlinks to external URLs that provide relevant information about the issue or a quick fix.
+
+[source,terminal]
+----
+links:
+  - url: "konveyor.io"
+    title: "Short title for the link"
+----
+
+Message can also be templated to include information about the match interpolated through custom variables on the rule.
```

Comment:
"A message"  "Templated" is a verb, as I just learned, but "used as a template" or "constructed as a template" (or something like these) might be easier to read. If you think it's best to use the verb, then keep it.